adapters/sn2core/sn2core.go

```diff
@@ -156,17 +157,34 @@ func AdaptTransaction(transaction *starknet.Transaction) (core.Transaction, erro
 
 func AdaptDeclareTransaction(t *starknet.Transaction) *core.DeclareTransaction {
 	return &core.DeclareTransaction{
-		TransactionHash:      t.Hash,
-		SenderAddress:        t.SenderAddress,
-		MaxFee:               t.MaxFee,
-		TransactionSignature: *t.Signature,
-		Nonce:                t.Nonce,
-		Version:              (*core.TransactionVersion)(t.Version),
-		ClassHash:            t.ClassHash,
-		CompiledClassHash:    t.CompiledClassHash,
+		TransactionHash:       t.Hash,
+		SenderAddress:         t.SenderAddress,
+		MaxFee:                t.MaxFee,
+		TransactionSignature:  *t.Signature,
+		Nonce:                 t.Nonce,
+		Version:               (*core.TransactionVersion)(t.Version),
+		ClassHash:             t.ClassHash,
+		CompiledClassHash:     t.CompiledClassHash,
+		ResourceBounds:        adaptResourceBounds(t.ResourceBounds),
+		Tip:                   t.Tip,
+		PaymasterData:         t.PaymasterData,
+		AccountDeploymentData: t.AccountDeploymentData,
+		NonceDAMode:           core.DataAvailabilityMode(t.NonceDAMode),
+		FeeDAMode:             core.DataAvailabilityMode(t.FeeDAMode),
```

Comment:
These cannot be casted directly between each other.

core:
``` golang

type DataAvailabilityMode uint32

const (
	DAModeL1 DataAvailabilityMode = iota + 1
	DAModeL2
)
```

starknet:
``` golang

type DataAvailabilityMode uint32

const (
	DAModeL1 DataAvailabilityMode = iota
	DAModeL2
)

```

let's both make them start at 0. This way, we can get rid of core.DataAvailabilityMode.Index() as well.

Reply:
Fixed this and got rid of `core.DataAvailabilityMode.index` in a later commit.

I can rebase this PR to make it easier to review.

Reply:
Ah, ok. I will just review the entire change set then.